langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/InvokeAsExternal.java

```diff
@@ -47,6 +48,7 @@ public static Object invokeAsExternal(Object func, Object[] args) {
             argList.add(arg);
             argList.add(true);
         }
-        return function.asyncCall(argList.toArray(), o -> o, METADATA);
+        return AsyncUtils.invokeFunctionPointerAsyncTransactional(function, null, METADATA,
```

Comment:
This will break if any other function use function.asyncCall. IMO We need to modify asyncCall to copy the trans action context